docs/testing/playwright/02-e2e-testing.md

```diff
@@ -0,0 +1,125 @@
+---
+title: End-to-End Testing
+sidebar_label: E2E Testing
+---
+
+When it comes to writing end-to-end tests using the Stencil Playwright adapter, the best advice we can give is to leverage
+[Playwright documentation](https://playwright.dev/docs/writing-tests). The adapter is set up in a way so that developers will use the
+same public APIs with as little Stencil nuances as possible.
+
+## Writing Tests
+
+As far as writing the actual tests goes, the most important thing to be aware of for tests to run correctly is to import the `test` function from
+`@stencil/playwright`, **not** directly from `@playwright/test`:
+
+```ts
+// THIS IS CORRECT
+import { test } from '@stencil/playwright';
+
+// THIS IS NOT!!
+// import { test } from '@playwright/test';
+```
+
+The adapter package extends Playwright's stock `test` function to provide additional fixtures and handle nuances related to web component hydration. More
+information is available in the [API documentation](./03-api.md#test-function).
+
+### Testing Patterns
+
+#### `page.goto()`
+
+The `goto()` method allows tests to load a pre-defined HTML template. This pattern is great if a test file has many tests to execute that all use the same HTML code
+or if additional `script` or `style` tags need to be included in the HTML. However, with this pattern, developers are responsible for defining the necessary `script`
+tags pointing to the Stencil entry code (so all web components are correctly loaded and registered).
+
+```html title="my-component.e2e.html"
+<!doctype html>
+<html lang="en">
+  <head>
+    <meta charset="utf8" />
+
+    <!-- Replace with the path to your entrypoint -->
```

Comment:
Do we know if folks will know that an 'entrypoint' is in this context? Is there anything we could do to clarify what 'test-app' should be in their project?